### DISCUSSION TITLE
Draft: Reworked Probabilities

### DISCUSSION THREAD

**Discussion by bnhrsch** [SynthstromAudible/DelugeFirmware#1476] [Ideas]
Right now the probability options are very basic (only percentage and iteration). To give a note probability you hold it down and turn the select knob. I'd like to propose a reworked Probability-Menu inspired by the Step Components of the OP-Z. Instead of turning the Select-knob you press it (while holding a note) to open up a menu with more options. 

**Use-cases :** 
- Create intricate and always evolving generative tracks that are not purely random, but follow a certain structure
- Spice up songs with ever changing but not random percussion or synth elements
- Randomly evolving textures without the need for long samples. 
- Create complex and evolving Percussion/Melodies without making long clips

**Reworked Probability:** 
Holding down a Note and pressing the Select opens up the Probability-Menu. You navigate it like every other menu by turning select and pressing select to enter and set a value. **One note can have multiple values active _(See Example below)_**

-------------------------

**Probability Menu Options:**

**Random Chance:** From 1-100% chance the note is played 

**Iteration Play:** Set iteration when the note is played (f.e. 1 every X bars)

**Iteration Skip:** Set iteration when the note is skipped (f.e. 1 every X bars)

**Mute:** (on/off): Mutes the Tone on that step (useful to use jump, hold, reverse without playing a tone)

**Pulse:** Repeats a step for a given number of steps (1 - 16). 
The playhead does not continue moving in that row until all pulse steps have been played

**Hold:** A note is held for a certain number of steps (1-16).
The playhead does not continue moving in that row until all steps have been held.

**Jump:** Jumps the playhead for this row forward or backward (+/- 1-16 steps)

**Reverse:** Chance of reversing the play-direction (percentage form 1-100%). If it triggers the play direction for that row is reversed. 

-------------------------

**Example 1: Percussion** 
A Note in a Percussive Kit is set to: 
-  Random Chance 25%
-  Pulse: 4
So there would be a 25% chance the note is triggered and a percussion Sample plays 4 times (the playhead does not move in that row as long as it plays). 

**Example 2: Generative Arp** 
An Arp-Synth is set to: 
- Iteration Play: 2 of 4
- Hold: 1/32th 
- Jump: -8
- Reverse: 50%  
The Synth-Arp starts playing in the second bar of four. It will hold for 1/32 and then jump the playhead for that row back 8 steps. After the Jump there is also a 50% chance that the play direction of that row is reversed. _Since you already can map the Arp-Rate and other values to LFOs you could create Semi-Generative Arps._ 

**Example 3: Generative Synth melodies** 
A Note in a Synth is set to: 
- Iteration Skip: 4 of 4
- Mute: On
- Jump +6
A note in a Synth is set to always trigger and only be skipped in the 4th bar of 4. If it triggers it will jump the play head 6 steps forward (skipping the next 5 notes after it in that row). It is also set to "Mute" so no tone is produced when it triggers. When the 4th bar is reached and the note is skipped no jumps will occur and the next 5 steps in that row are played normally. _You could use this in multiple rows with different Values to create always evolving generative melodies._

**Comment by DW-James** [comment]
Yes to this! The Five12 Vector Sequencer and the Numerology software allow for this type of sequencing and it's wonderful how interesting things can get with only a few of these rules in place.  
I think it works well when each function (SKIP, REPEAT, OCTAVE UP, NOTE +/- X SEMITONES, etc) has its own probability of that thing firing. 

I would caveat by saying that I found with the Vector that it's important to have a RESET parameter for the track/clip every X beats/bars when using functions like this to be able to dial in that sweet spot of being predictable yet interesting and evolving musically and not so random that you lose interest.

**Comment by sndnmd** [comment]
how about adding a velocity offset chance? so it can have a chance of beeing +/- 10 velocity. the opz had that if i remember right
